### ISSUE TITLE
Implement dynamic nav menu

### ISSUE THREAD

**Issue by Michiel-s** [AmpersandTarski/prototype-frontend#164]
## Context
The backend API provides information that allows for dynamic navigation menu. Instead of hard coding the navigation menu in the frontend, we can load the menu structure from the API. The menu structure is stored in the database according to an Ampersand model, which can be changed in the frontend itself using a (generated) interface.

## Nav menu structure from the API
When calling `GET api/v1/app/navbar` you'll get a data structure below. Notice the `navs` field; an array of `PF_MenuItem`s.

```ts
type NavBarInterface = {
  defaultSettings: PF_Settings;
  home: string;
  session: {
    id: SessionId;
    loggedIn: boolean;
  };
  sessionRoles: Array<PF_Role>;
  sessionVars: false | { [key: string]: unknown };
  notifications: Notifications;
  navs: Array<PF_MenuItem>;
  new: Array<PF_MenuItemNew>;
  ext: Array<unknown>; // ignore; not reusing in new frontend
  role: Array<unknown>; // ignore; not reusing in new frontend. This was structure for additional items in the role menu
};

type SessionId = string;

type PF_MenuItem = {
  id: PF_MenuItemId;
  label: string;
  ifc: string | null;
  parent: PF_MenuItemId | null;
  seqNr: number | null;
  url: string | null;
};
```

## Four kinds of menu items
A menu item can come in X different variations:
1. with `ifc` attribute specified; this is a direct link to an interface that can be navigated to
2. `ifc=null`, but with `url` specified; this is a (relative) URL to navigate to or external url (starting with http/https)
3. `ifc=null`, `url=null`, but other menu items refer to this item as parent; menu item that can be expended to view sub items. This applies to e.g. the root item (MainMenu).
4. none of the above. let's consider this one as seperator; a hairline with small text 

## Example data
An example of the `navs` structure looks like so:
```json
[
  {
    "id": "MainMenu",
    "label": "Main menu",
    "seqNr": null,
    "ifc": null,
    "url": null,
    "parent": null
  },
  {
    "id": "Active_32_projects",
    "label": "Active projects",
    "seqNr": 1,
    "ifc": "Active_32_projects",
    "url": null,
    "parent": "MainMenu"
  },
  {
    "id": "Edit_32_navigation_32_menu",
    "label": "Edit navigation menu",
    "seqNr": 2,
    "ifc": "Edit_32_navigation_32_menu",
    "url": null,
    "parent": "MainMenu"
  },
  {
    "id": "List_32_all_32_interfaces",
    "label": "List all interfaces",
    "seqNr": 4,
    "ifc": "List_32_all_32_interfaces",
    "url": null,
    "parent": "MainMenu"
  },
  {
    "id": "Inactive_32_projects",
    "label": "Inactive projects",
    "seqNr": 3,
    "ifc": "Inactive_32_projects",
    "url": null,
    "parent": "MainMenu"
  },
  {
    "id": "People",
    "label": "People",
    "seqNr": 6,
    "ifc": "People",
    "url": null,
    "parent": "MainMenu"
  },
  {
    "id": "My_32_projects",
    "label": "My projects",
    "seqNr": 5,
    "ifc": "My_32_projects",
    "url": null,
    "parent": "MainMenu"
  }
]
```

## Implementation in the frontend
The menu section `Project Administration` with project specific sub menu items that is currently hard codes can be replaced by the MainMenu from the API.

![image](https://user-images.githubusercontent.com/7961582/216821274-94a32bc0-6e3d-402e-8d59-181056332155.png)

## Nested menu items
Note that menu items can have sub items (recursively). That needs to be supported. The [primeng menu model](https://primeng.org/menumodel) that we use has support for this. Same applies to the [Sakai layout](https://www.primefaces.org/sakai-ng/#/) we use. Example below:

![image](https://user-images.githubusercontent.com/7961582/216828583-69e451c9-e811-463d-a085-0dcfb93f0a84.png)

## Multiple top level items must be possible
Note that it is possible to have multiple root items. That should result in an additional menu section. Like this:

![image](https://user-images.githubusercontent.com/7961582/216833331-81b1f229-3a1a-43e2-b12b-2fecdbd5a9d1.png)

## Static items
The two top level menu items `PROTOTYPE` and `ADMIN` can be kept (for now). We'll look at that in another issue.

**Comment by byTinus** [comment]
@Michiel-s If a menuItem has a parent assigned, say `Item10`, can we assume that `Item10` exists among the other menuItems? 

I plan on writing a loop that tries to add items with a parent assigned, to their parent. We can only parent an item to an existing object, and it can happen that a submenu appears higher in the API response than its parent, i.e. is added before its parent. Thus I made a temporary `childItems()` array to park unassigned submenus. Once the other menuItems are added, it loops through `childItems()` until they are all added. To prevent an infinite loop, it is important to know for certain if assigned parent ids exist.

**Comment by hanjoosten** [comment]
Yes, the generator can guarantee this. However, it shouldn't hurt to test for this. If it doesn't exist, a hard crash (preferably with descent logging) would be nice. @Michiel-s , what do you think?
